test/new_relic/agent/log_event_aggregator_test.rb

```diff
@@ -0,0 +1,270 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'test_helper'))
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'data_container_tests'))
+
+require 'new_relic/agent/log_event_aggregator'
+
+module NewRelic::Agent
+  class LogEventAggregatorTest < Minitest::Test
+    def setup
+      nr_freeze_process_time
+      @aggregator = NewRelic::Agent.agent.log_event_aggregator
+      @aggregator.reset!
+
+      @enabled_config = { LogEventAggregator::FORWARDING_ENABLED_KEY => true }
+      NewRelic::Agent.config.add_config_for_testing(@enabled_config)
+
+      # Callbacks for enabled only happen on SSC addition
+      NewRelic::Agent.config.notify_server_source_added
+
+      NewRelic::Agent.instance.stats_engine.reset!
+    end
+
+    def teardown
+      NewRelic::Agent.config.remove_config(@enabled_config)
+    end
+
+    CAPACITY_KEY = LogEventAggregator.capacity_key
+
+    # Helpers for DataContainerTests
```

Comment:
DataContainerTests are new to me, and I don't believe I quite understand their purpose by reading `test/new_relic/data_container_tests.rb`. What constitutes a DataContainer? When should one be added? (also happy to discuss these during a synchronous meeting if that's easier)

Reply:
So a `DataContainer` is the abstraction for "something that buffers data until harvest". The various event aggregators, error collectors and metric takers are all examples.

These "containers" get used somewhat generically by parts of the `NewRelicService` during harvest. The `DataContainerTests` module defines a set of test methods that captures those generic behaviors, largely around required methods, resets and merging items back upon failure.